nursery/coin_select/src/coin_selector.rs

```diff
@@ -1,173 +1,171 @@
 use super::*;
+#[allow(unused)] // some bug in <= 1.48.0 sees this as unused when it isn't
+use crate::float::FloatExt;
+use crate::{bnb::BnBMetric, float::Ordf32, FeeRate};
+use alloc::{borrow::Cow, collections::BTreeSet, vec::Vec};
+
+/// [`CoinSelector`] is responsible for selecting and deselecting from a set of canididates.
+///
+/// You can do this manually by calling methods like [`select`] or automatically with methods like [`branch_and_bound`].
+///
+/// [`select`]: CoinSelector::select
+/// [`branch_and_bound`]: CoinSelector::branch_and_bound
+#[derive(Debug, Clone)]
+pub struct CoinSelector<'a> {
+    base_weight: u32,
+    candidates: &'a [Candidate],
```

Comment:
I think it will be much more helpful if we had:
```rust
pub struct CoinSelector<'c, C> {
    candidates: &'c [C],
    /* ... */
}

impl<'c, C: AsRef<Candidate>> CoinSelector<'c, C> {}
```

This way `C` can contain everything required to complete the rest of tx building.

Reply:
I think to do this properly we'd have to make `Candidate` a trait. I'd leave it like this for now until we see how tx building turns out and make it more flexible where we find pain points.